packages/native/src/useLinking.tsx

```diff
@@ -92,10 +104,10 @@ export default function useLinking(
   const numberOfIndicesAhead = React.useRef(0);
 
   React.useEffect(() => {
-    window.addEventListener('popstate', () => {
+    const onPopState = () => {
```

Comment:
Maybe it would make sense to split this into a new hook `usePopState` to keep the logic divided more clearly 🤔

Reply:
I'll refactor after adding some tests so I don't accidentally break anything 😅